infra.bs

```diff
@@ -608,58 +608,58 @@ U+007A (z), inclusive.
 <a>scalar value</a> and maps isolated surrogates to their corresponding <a>code point</a>, leaving
 them effectively as-is.
 
-<p class=example id=example-javascript-string-in-code-points>A <a>JavaScript string</a> consisting
-of the <a>code units</a> 0xD83D, 0xDCA9, and 0xD800, when interpreted as containing
-<a>code points</a>, would consist of the <a>code points</a> U+1F4A9 and U+D800.
+<p class=example id=example-javascript-string-in-code-points>A <a>string</a> consisting of the
+<a>code units</a> 0xD83D, 0xDCA9, and 0xD800, when interpreted as containing <a>code points</a>,
+would consist of the <a>code points</a> U+1F4A9 and U+D800.
+
+<p>A <a>string</a>'s
+<dfn export for="string,JavaScript string,scalar value string" id=javascript-string-length>length</dfn>
+is the number of <a>code units</a> it contains.
+
+<p>A <a>string</a>'s
+<dfn export for="string,JavaScript string,scalar value string">code point length</dfn> is the number
+of <a>code points</a> it contains.
 
-<p>A <dfn export>scalar value string</dfn> is a sequence of <a>scalar values</a>.
+<p>A <dfn export>scalar value string</dfn> is a <a>string</a> whose <a>code points</a> are all
+<a>scalar values</a>.
 
 <p class=note>A <a>scalar value string</a> is useful for any kind of I/O or other kind of operation
 where <a>UTF-8 encode</a> comes into play.
 <!-- It's also useful if you can imagine the subsystem to be implemented in Rust -->
 
-<p><dfn export lt=string>String</dfn> can be used to refer to either a <a>JavaScript string</a> or
-<a>scalar value string</a>, when it is clear from the context which is meant or when the distinction
-is immaterial. <a>Strings</a> are denoted by double quotes and monospace font.
+<p>To <dfn export for="string,JavaScript string" id=javascript-string-convert>convert</dfn> a
+<a>string</a> into a <a>scalar value string</a>, replace any <a>surrogates</a> with U+FFFD.
 
-<p class=example id=example-string-notation>"<code>Hello, world!</code>" is a string.
-
-<p>A <a>string</a>'s <dfn export for=string>length</dfn> is the number of <a>code points</a> it
-contains.
-
-<p>To <dfn export for="JavaScript string">convert</dfn> a <a>JavaScript string</a> into a
-<a>scalar value string</a>, replace any <a>surrogates</a> with U+FFFD.
-<!-- Obviates need for https://heycam.github.io/webidl/#dfn-obtain-unicode -->
+<div class=note>
+ <p>The replaced surrogates are always isolated surrogates, since the process of interpreting the
+ string as containing <a>code points</a> will have converted surrogate pairs into
+ <a>scalar values</a>.
 
-<p class=note>The replaced surrogates are always isolated surrogates, since the process of
-interpreting the JavaScript string as containing <a>code points</a> will have converted surrogate
-pairs into <a>scalar values</a>.
+ <p>A <a>scalar value string</a> can always be used as a <a>string</a> implicitly since it is a
+ subset. The reverse is only possible if the <a>string</a> is known to not contain
```

Comment:
Preexisting issue, but "it is a subset" wording doesn't seem quite right. "A scalar value string" is not a subset of "a string", which is what this grammatical structure implies. Potential rephrasings:

- "since every scalar value string is a string"
- "since the set of all scalar value strings is a subset of the set of all strings"